shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -206,17 +223,82 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
 
   /**
    * Generates the configuration docs.
+   * @param validate validate flag
    */
-  public static void generate() throws IOException {
+  public static void generate(boolean validate) throws IOException {
     Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
     defaultKeys.removeIf(key -> key.isHidden());
     String homeDir = new InstancedConfiguration(ConfigurationUtils.defaults())
         .get(PropertyKey.HOME);
     // generate CSV files
     String filePath = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
-    writeCSVFile(defaultKeys, filePath);
+    writeCSVFile(defaultKeys, filePath, validate);
+
+    if (validate) {
```

Comment:
since you have all the info you need within the writeCSVFile and writeYAMLFile functions, i think the validation logic should also belong within each respective function. since they would be handling the write and optionally the validate step, i would also rename them from write*File to handle*File